src/nrfx/hal/nrf_egu.c

```diff
@@ -0,0 +1,136 @@
+/** Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Note that the function prototypes are taken from the NRFx HAL
+ */
+
+#include "hal/nrf_egu.h"
+#include "bs_tracing.h"
+#include "NRF_EGU.h"
+#include <kernel.h>
+#include <irq_ctrl.h>
+
+uint32_t nrf_egu_channel_count(NRF_EGU_Type const *p_reg)
+{
+	return sizeof(p_reg->EVENTS_TRIGGERED) / sizeof(p_reg->EVENTS_TRIGGERED[0]);
+}
+
+void nrf_egu_task_trigger(NRF_EGU_Type *p_reg, nrf_egu_task_t egu_task)
+{
+	*((volatile uint32_t *)((uint8_t *)p_reg + (uint32_t)egu_task)) = 0x1UL;
+	nrf_egu_regw_sideeffects_TASKS_TRIGGER(p_reg);
+	nrf_ppi_event(PPI_EGU_EVENTS[REG_TO_ID(p_reg)][TASK_TO_ID(egu_task)]);
+}
+
+uint32_t nrf_egu_task_address_get(NRF_EGU_Type const *p_reg,
+				  nrf_egu_task_t egu_task)
+{
+    return (uint32_t)((uint8_t *)p_reg + (uint32_t)egu_task);
+}
+
+nrf_egu_task_t nrf_egu_trigger_task_get(uint8_t channel)
+{
+	return (nrf_egu_task_t)offsetof(NRF_EGU_Type, TASKS_TRIGGER[channel]);
+}
+
+bool nrf_egu_event_check(NRF_EGU_Type const *p_reg, nrf_egu_event_t egu_event)
+{
+	return *((volatile uint32_t *)((uint8_t *)p_reg + (uint32_t)egu_event));
+}
+
+void nrf_egu_event_clear(NRF_EGU_Type *p_reg, nrf_egu_event_t egu_event)
+{
+    *((volatile uint32_t *)((uint8_t *)p_reg + (uint32_t)egu_event)) = 0x0UL;
```

Comment:
Minor: magic number

Reply:
IMHO 0 and 1 in this context is acceptable